frank/make_figs.py

```diff
@@ -454,23 +604,340 @@ def make_diag_fig(r, q, iteration_diagnostics, iter_plot_range=None,
+    clean_DHT_kl = sol.predict_deprojected(grid,
+                                           I=np.interp(DHT.r, r_clean, I_clean)).real * 1e3
+
+    plot_vis_fit(grid, vis_fit_kl, ax1, c='r', label='frank>0')
+    plot_vis_fit(grid, -vis_fit_kl, ax1, c='r', ls='--', label='frank<0')
+    plot_vis_fit(grid, clean_DHT_kl, ax1, c='b', label='DHT of CLEAN>0')
+    plot_vis_fit(grid, -clean_DHT_kl, ax1, c='b', ls='--', label='DHT of CLEAN<0')
+
+    vmin = 0
+    if mean_convolved is not None:
+        vmax = max(sol.mean.max(), mean_convolved.max(), I_clean.max())
+    else:
+        vmax = max(sol.mean.max(), I_clean.max())
+    norm = PowerNorm(gamma, 0, vmax)
+
+    plot_2dsweep(sol.r, sol.mean, ax=ax2, cmap='inferno', norm=norm, vmin=0,
+                vmax=vmax / 1e10, xmax=sol.Rmax, plot_colorbar=True)
+    if mean_convolved is not None:
+        plot_2dsweep(sol.r, mean_convolved, ax=ax3, cmap='inferno', norm=norm,
+                    vmin=0, vmax=vmax / 1e10, xmax=sol.Rmax, plot_colorbar=True)
+
+    # Interpolate the CLEAN profile onto the frank grid to ensure the CLEAN
+    # swept 'image' has the same pixel resolution as the frank swept 'images'
+    from scipy.interpolate import interp1d
+    interp = interp1d(r_clean, I_clean)
+    regrid_I_clean = interp(sol.r)
+    plot_2dsweep(sol.r, regrid_I_clean, ax=ax4, cmap='inferno', norm=norm,
+                vmin=0, vmax=vmax / 1e10, xmax=sol.Rmax, plot_colorbar=True)
+
+    ax0.legend(loc='best')
+    ax1.legend(loc='best')
+
+    ax0.set_xlabel('r ["]')
+    ax0.set_ylabel(r'Brightness [$10^{10}$ Jy sr$^{-1}$]')
+    ax1.set_xlabel(r'Baseline [$\lambda$]')
+    ax1.set_ylabel(r'Re(V) [mJy]')
+    ax2.set_xlabel('RA offset ["]')
+    ax3.set_xlabel('RA offset ["]')
+    ax4.set_xlabel('RA offset ["]')
+    ax2.set_ylabel('Dec offset ["]')
+
+    ax0.set_xlim(right=sol.Rmax)
+    if dist:
+        xlims = ax0.get_xlim()
+        ax0_5.set_xlim(np.multiply(xlims, dist))
+
+    ax1.set_xlim(.9 * baselines.min(), 1.2 * baselines.max())
+    ax1.set_xscale('log')
+    ax1.set_yscale('log')
+    ax1.set_ylim(bottom=1e-3)
+
+    ax2.set_title('Unconvolved frank profile swept')
+    ax3.set_title('Convolved frank profile swept')
+    ax4.set_title('CLEAN profile swept')
+
+    ax0.text(.5, .9, 'a)', transform=ax0.transAxes)
+    ax1.text(.5, .9, 'b)', transform=ax1.transAxes)
+    ax2.text(.1, .9, 'c)', c='w', transform=ax2.transAxes)
+    ax3.text(.1, .9, 'd)', c='w', transform=ax3.transAxes)
+    ax4.text(.1, .9, 'e)', c='w', transform=ax4.transAxes)
+
+    if save_prefix:
+        plt.savefig(save_prefix + '_frank_clean_comparison.png', dpi=600)
+        plt.close()
+    else:
+        plt.show()
+
+    return fig, axes
+
+
+def make_multifit_fig(u, v, vis, weights, sols, bin_widths, varied_pars,
+                      varied_vals, dist=None, force_style=True, save_prefix=None
+                     ):
+    r"""
+    Produce a figure overplotting multiple fits
+
```

Comment:
Yes make_quick_fig shows the brightness profile in lin and log, the visibility fit in lin, the visibility residuals in lin.

make_multifit_fig shows the brightness profile in lin and log, the visibility fit in lin _and log_ (because often the difference between fits isn't distinguishable in lin), the power spectrum.

I'd like to keep them separate because if make_quick_fig shows the power spectrum, it should also show the log visibility fit to see how similar the structure in the power spectrum is (at that point it's better to call make_full_fig); and the overplot fig doesn't need to compare the visibility residuals, which are a coarse metric of the fit anyway (especially for multiple fits that are going out to different maximum baselines).

Reply:
Ok, that's fine